### PR TITLE
feat: generate & deconstruct of draw-pattern

### PR DESCRIPTION
**Pull request by BQXBQX**
# Generate & deconstruct of studio-draw-pattern

## Change
Added a studio-draw-pattern in the packages to achieve two-way conversion of query statements.

## State
Currently, by designing a set of JSON, we have realized the bidirectional conversion of Cypher's MATCH statement, using regular expressions for the forward direction and string concatenation for the reverse direction.

## Notice
The code is currently under development, developed using Bun and typescript, and the React version has not been developed or optimized. Stay tuned.

### CODE REVIEW

File: packages/studio-draw-pattern/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "generate_query",
```
Comment (pomelo-nwu) [inline packages/studio-draw-pattern/package.json:2]: name 可以为 `@graphscope/studio-draw-pattern`

File: packages/studio-draw-pattern/types/node.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { Property } from "./property";
+
+export interface Node {
```
Comment (pomelo-nwu) [inline packages/studio-draw-pattern/types/node.d.ts:3]: 补充下代码注释，方便后续维护

File: packages/studio-draw-pattern/utils/cypher/generateMATCH.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
+import type { Edge } from "../../types/edge";
```
Comment (pomelo-nwu) [inline packages/studio-draw-pattern/utils/cypher/generateMATCH.ts:1]: 逻辑之后再详细 review

File: packages/studio-draw-pattern/test/hasIntersection.test.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import hasIntersection from "../utils/public_functions/hasIntersection";
+
+const array1 = [1, 2, 3];
+const array2 = [1, 4, 5];
+
+test("array1 and array2 intersection", () => {
```
Comment (pomelo-nwu) [inline packages/studio-draw-pattern/test/hasIntersection.test.ts:6]: 完善下单元测试

**Comment by pomelo-nwu** [comment]
- related：https://github.com/alibaba/GraphScope/issues/3740

**Comment by BQXBQX** [comment]
# Approaching
Refactoring the code to adapt to react. The plan is to expose two React Hooks and one React Component for better web integration.

## Develop Tools
- 📦 Build: Turbopack
- 🏃‍♂️ State：Valtio

## Productions
- React Hook: useGenerate
- React Hook: useDeconstruct
- React Component: DrawPattern

**Pull request closed without merging**